cube_holder.scad

```diff
@@ -17,29 +17,39 @@ cube_holder();
 // Begin modules that can be used to assemble this piece in solver_3.scad
 
 module above_cube_holder() {
-  up(thickness/2 + servo_gear_thickness) above_frame() children();
+  up(servo_gear_thickness) above_frame() children();
 }
 
 module cube_holder() {
+  up(servo_gear_thickness) onto_point()
+    piece_sleeve();
+  holder_gear();
+}
+
+module piece_sleeve() {
   piece_size = cube_size / 3 + margin;
   holder_open_size = piece_size + thickness;
   holder_closed_size = piece_size + thickness + embedded_thickness;
-  difference() {
-    up(servo_gear_thickness - thickness/2)
-      onto_point(holder_closed_size)
-      difference() {
+  translate([-piece_size/2,piece_size/2,piece_size/2]) {
+    difference() {
+      translate([thickness/2, (embedded_thickness-thickness)/2, (embedded_thickness-thickness)/2])
+        ccube([holder_open_size, holder_closed_size, holder_closed_size], x+y+z);
+      rounded_cube(piece_size, cube_rounding, x+y+z);
+      translate([-thickness, 0, 0]) rounded_cube(piece_size, cube_rounding, x+y+z);
+      rotate(45,[1,0,1])
+        translate([0,holder_open_size,0])
         ccube([holder_open_size, holder_closed_size, holder_closed_size], x+y+z);
-        translate([-thickness/2, thickness/2, thickness/2]) {
-          rounded_cube(piece_size, cube_rounding, x+y+z);
-          translate([-thickness, 0, 0]) rounded_cube(piece_size, cube_rounding, x+y+z);
-          //translate([-cube_rounding, cube_rounding, cube_rounding]) ccube(piece_size, x+y+z);
-        }
     }
-    up(piece_size * 2) ccube(piece_size * 3, x + y);
+    translate([(piece_size + thickness)/2,-(piece_size/2 + thickness),(piece_size + thickness)/-2])
+      rotate(90,[1,0,0])
+      rotate(22.5,[0,0,1])
+      pin(thickness*2.5,thickness/2);
   }
-  holder_gear();
 }
 
 module holder_gear() {
-  gear(servo_gear_teeth_size, servo_gear_num_teeth*2, servo_gear_thickness);
+  difference() {
+    gear(servo_gear_teeth_size, servo_gear_num_teeth*4, servo_gear_thickness);
+    up(servo_gear_thickness) onto_point() piece_sleeve();
```

Comment:
Since piece_sleeve() is hollow, this might not cut out everything in needs to.

In this case, I usually add a `hollow=true` param to the module in question (in this case `piece_sleeve`) and then in my `difference()` functions, do `difference() { foo(); if(hollow) { baz(); qux(); } }`

That way, here you can cut out `piece_sleeve(hollow=false)` and be sure that the whole body of the piece sleeve is cut out of the gear, regardless of `thickness`, `piece_size`, etc